py/server/deephaven/dbc/adbc.py

```diff
@@ -0,0 +1,45 @@
+#
+#     Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+#
+"""This module supports ingesting data from external databases (relational and other types) into Deephaven via the
+Python DB-API 2.0 ( PEP 249) and the Apache Arrow Database Connectivity (ADBC) interfaces. (Please refer to
+https://arrow.apache.org/docs/dev/format/ADBC.html for more details on ADBC).
+
+ADBC defines a standard API to fetch data in Arrow format from databases that support Arrow natively as well as
+from databases that only support ODBC/JDBC. By relying on ADBC, Deephaven is able to ingest data efficiently from a
+wide variety of data sources. """
+from typing import Any
+
+from deephaven.table import Table
+from deephaven import arrow as dharrow
+from deephaven import DHError
+
+try:
+    import adbc_driver_manager.dbapi
+except ImportError:
+    raise DHError(message="import ADBC driver manager failed")
+
+
+def read_cursor(cursor: Any) -> Table:
```

Comment:
If we are going this way of splitting dbc types into files, we should really be doing proper typing for cursors.